mri/test/test_optimizer.py

```diff
@@ -84,6 +85,47 @@ def test_reconstruction_fista_fft2(self):
                     np.testing.assert_allclose(x_final, np.fft.ifftshift(
                         fourier_0.adj_op(data_0)))
 
+    def test_reconstruction_pogm_fft2(self):
+        """ Test all the registered transformations.
+        """
+        print("Process test FFT2 POGM::")
+        for image in self.images:
+            fourier = FFT2(samples=convert_mask_to_locations(
+                                            fftshift(self.mask)),
+                           shape=image.shape)
+
+            data = fourier.op(image.data)
+            print("Process test with image '{0}'...".format(
+                image.metadata["path"]))
+            for nb_scale in self.nb_scales:
+                print("- Number of scales: {0}".format(nb_scale))
+                for name in self.decimated_wavelets:
+                    print("    Transform: {0}".format(name))
+                    gradient_op, linear_op, prox_op, cost_op = \
+                        generate_operators(
+                            data=data,
+                            wavelet_name=name,
+                            samples=convert_mask_to_locations(
+                                fftshift(self.mask)),
+                            mu=0,
+                            nb_scales=4,
+                            non_cartesian=False,
+                            uniform_data_shape=image.shape,
+                            gradient_space="synthesis")
+                    x_final, costs, _ = sparse_rec_pogm(
+                        gradient_op=gradient_op,
+                        linear_op=linear_op,
+                        prox_op=prox_op,
+                        cost_op=cost_op,
+                        max_nb_of_iter=self.nb_iter,
+                        verbose=0)
+                    fourier_0 = FFT2(samples=convert_mask_to_locations(
+                                            fftshift(self.mask)),
+                                     shape=image.shape)
+                    data_0 = fourier_0.op(np.fft.fftshift(image.data))
+                    np.testing.assert_allclose(x_final, np.fft.ifftshift(
+                        fourier_0.adj_op(data_0)))
```

Comment:
All these ffshifts are making me nervous. I don't understand why they are not handled directly in the FFT objects we have.

Reply:
Resolving as a duplicate of above